src/PKSim.Core/Snapshots/Mappers/ClassificationMapper.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ModelClassification = OSPSuite.Core.Domain.Classification;
+using SnapshotClassification = PKSim.Core.Snapshots.Classification;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public class ClassificationMapper : SnapshotMapperBase<ModelClassification, SnapshotClassification, object, ClassificationContext> 
+   {
+      public override Task<ModelClassification> MapToModel(SnapshotClassification snapshot, object context)
+      {
+         throw new NotImplementedException();
+      }
+
+      protected async Task<SnapshotClassification> MapClassificationToSnapshot(ModelClassification model)
+      {
+         if (model == null)
+            return null;
+
+         var snapshot = await SnapshotFrom(model, x =>
+         {
+            x.Name = model.Name;
+         });
+         return snapshot;
+      }
+
+      public override async Task<SnapshotClassification> MapToSnapshot(ModelClassification model, ClassificationContext context)
+      {
+         return await MapTreeFrom(model, context.Classifications, context.Classifiables);
+      }
+
+      protected async Task<SnapshotClassification> MapTreeFrom(ModelClassification classification, IReadOnlyList<ModelClassification> classifications, IReadOnlyList<ClassifiableContext> classifiables)
+      {
+         var root = await MapClassificationToSnapshot(classification);
+
+         var childClassifications = childClassificationsFrom(classification, classifications);
+         await mapChildClassifications(classifications, classifiables, childClassifications, root);
```

Comment:
it it possible to assign classification to root instead of passing root as parameter (avoiding thus side effects)
sthg like
root.Classifications  =await ... 

Looks more like the next line